src/collection.ts

```diff
@@ -235,44 +213,16 @@ export class Collection<T extends TigrisCollectionType> {
 	}
 
 	findMany(
-		filter: SelectorFilter<T> | LogicalFilter<T> | Selector<T>,
+		filter?: SelectorFilter<T> | LogicalFilter<T> | Selector<T>,
 		readFields?: ReadFields,
 		tx?: Session,
 		options?: ReadRequestOptions
-	): Promise<Array<T>> {
-		return new Promise<Array<T>>((resolve, reject) => {
-			if (options === undefined) {
-				options = new ReadRequestOptions();
-			}
-			const result: Array<T> = new Array<T>();
-			this.findManyStream(
-				filter,
-				{
-					onEnd() {
-						resolve(result);
-					},
-					onNext(item: T) {
-						result.push(item);
-					},
-					// eslint-disable-next-line @typescript-eslint/no-unused-vars
-					onError(_error: Error) {
-						reject(_error);
-					},
-				},
-				readFields,
-				tx,
-				options
-			);
-		});
-	}
+	): FindCursor<T> {
```

Comment:
rename this to just `Cursor`

Reply:
What are the feature specific to find cursor that doesn't apply to search cusror?

Reply:
FindCursor can expose skip/limit based pagination (if we eventually decide to). But Search uses page/pageSize based pagination. What would just returning top level interface buy us here? If its for extraction, we could as well treat FindCursor as the interface for find queries.

Reply:
Could we call this `Cursor` and then for search we could call it `SearchCursor`. The name `FindCursor` looks a little odd.

Reply:
Cool. So Cursor and AbstractCursor?